packages/ChangesReloaded-Core.package/CRChangeListItem.class/instance/className.st

```diff
@@ -0,0 +1,6 @@
+accessing
+className
+
+	self isMethod ifTrue: [^ self methodClassName].
+	self isClassComment ifTrue: [^ self commentClass name].
+	^ 'undefined'
```

Comment:
Is this intended to be reachable? If so, is there a benefit of returning `'undefined'` over returning `nil`?

I usually have the principle to be as error-throwy as possible when encountering unexpected situations or inputs. If this control flow branch is never intended to be reached, I'd throw an error instead – this shows to developers that they called it wrong or created the `CRChangeListItem` in a wrong way.
If this control flow branch is intended to be reached, I would still rather return `nil` so as to show that this is a special case that needs to be handled separately.
Otherwise, returning `'undefined'` if there's no class name _is part of the API_ of the `CRChangeListItem`. That means in other parts of the code base, you might need to write `className == 'undefined' ifTrue: […]`, which I consider quite ugly.

Reply:
fair